motionplan/motionPlanner.go

```diff
@@ -62,68 +68,92 @@ func (req *PlanRequest) validatePlanRequest() error {
 	if req.Logger == nil {
 		return errors.New("PlanRequest cannot have nil logger")
 	}
-	if req.Frame == nil {
-		return errors.New("PlanRequest cannot have nil frame")
-	}
 
 	if req.FrameSystem == nil {
 		return errors.New("PlanRequest cannot have nil framesystem")
-	} else if req.FrameSystem.Frame(req.Frame.Name()) == nil {
-		return referenceframe.NewFrameMissingError(req.Frame.Name())
 	}
-
-	if req.Goal == nil {
-		return errors.New("PlanRequest cannot have nil goal")
+	if req.StartState == nil {
+		return errors.New("PlanRequest cannot have nil StartState")
 	}
-
-	goalParentFrame := req.Goal.Parent()
-	if req.FrameSystem.Frame(goalParentFrame) == nil {
-		return referenceframe.NewParentFrameMissingError(req.Goal.Name(), goalParentFrame)
+	if req.StartState.configuration == nil {
+		return errors.New("PlanRequest cannot have nil StartState configuration")
 	}
-
-	if len(req.BoundingRegions) > 0 {
-		buffer, ok := req.Options["collision_buffer_mm"].(float64)
-		if !ok {
-			buffer = defaultCollisionBufferMM
-		}
-		// check that the request frame's geometries are within or in collision with the bounding regions
-		robotGifs, err := req.Frame.Geometries(make([]referenceframe.Input, len(req.Frame.DoF())))
-		if err != nil {
-			return err
+	// If we have a start configuration, check for correctness
+	for fName, seed := range req.StartState.configuration {
```

Comment:
I have wp.ComputePoses, however, converting to a PathStep isn't what we are doing here.

PlanState is just a convenient data structure; it's used differently when passed publicly as a goal vs how others are constructed as private members of private structs, so a more broad validate function would not be appropriate.

Reply:
Yeah as I read on I found ComputePoses and feel like thats almost what you want to just be calling here.

Reply:
Sorry I wasn't very clear with my second comment.  I think a Transform method on Framesystem would be a good method to have to convert a Configuration to a PathStep.  I understand you aren't using that path step here, but we do this elsewhere in the codebase where we transform a model by an Input, throw away the result and check the error to see if the calculation succeeded.  I think a corrolary to this operating on the Framesystem would be useful not only here but elsewhere as we move away from Inputs and towards PlanStates as the atomic data type

Reply:
This lets us not have iterate over frames and check their DOF which should be the purview of the referenceframe package and not the motionplan package.  Its the blurring of the abstractions that we've established that I am hoping to correct

Reply:
Oh, I see. This makes sense, I'll take a stab at this.

Reply:
Unfortunately it's not possible for `referenceframe` to have a configuration -> pathstep function, since pathstep is in motionplan, at the moment. But that validation can indeed happen just one place.